### ISSUE TITLE
Problem with 'truffle test'

### ISSUE THREAD

**Issue by plainfatboy** [trufflesuite/truffle#4608]
- [X] I've [opened a support ticket](https://trufflesuite.zendesk.com/hc/en-us/requests/new) before filing this issue.

---------------------------

## Issue

I was following the tutorial ( Truffle Suite - Truffle Suite ). After running 'truffle test', there are characters raining down like the following video (https://www.reddit.com/link/rzv2m3/video/poupd8ewsoa81/player).

I found out a kind of warning "LIBERTY LIBERTY LIBERTY" before the characters raining as following snapshot (https://i.redd.it/q70lk747toa81.jpg).

## Steps to Reproduce

1. Follow the entire tutorial
2. Run 'truflle test'

## Expected Behavior

`Using network 'development'.

Compiling your contracts...
===========================
> Compiling ./test/TestAdoption.sol
> Artifacts written to /var/folders/z3/v0sd04ys11q2sh8tq38mz30c0000gn/T/test-11934-19747-g49sra.0ncrr
> Compiled successfully using:
   - solc: 0.5.0+commit.1d4f565a.Emscripten.clang

  TestAdoption
    ✓ testUserCanAdoptPet (91ms)
    ✓ testGetAdopterAddressByPetId (70ms)
    ✓ testGetAdopterAddressByPetIdInArray (89ms)

  3 passing (670ms)`

## Actual Results

Raining characters as shown in the video (https://www.reddit.com/link/rzv2m3/video/poupd8ewsoa81/player)

## Environment

* Operating System: Windows 11
* Ethereum client: Ganache
* Truffle version (`Truffle v5.4.27 (core: 5.4.27)`):
* node version (`Node v16.13.1`):
* npm version (`8.1.2`):

**Comment by plainfatboy** [comment]
This is duplicated with https://github.com/trufflesuite/truffle/issues/4607.

**Comment by plainfatboy** [comment]
I tried their solution but it does not work. The reason is they are testing using Javascript. However, I test the smart contract using Solidity.

**Comment by JoepdeJong** [comment]
> I tried their solution but it does not work. The reason is they are testing using Javascript. However, I test the smart contract using Solidity.

How do you run your tests? It works for me with a combination of `.sol` and `.js` test files.

The `.sol` tests work when running tests using the method described in #4607

You could also try to run the test as follows:
```
npx truffle develop
truffle(develop)> test
...
```

Hopefully, this helps. Could you otherwise tell me which commands you use to run your scripts?

**Comment by plainfatboy** [comment]
Thank you very much for your reply. I think I might know what could go wrong. I did not have `.js` test files in my project. Do we have to use a combination of `.sol` and `.js` test files for testing the smart contract?

Firstly, I tried
```
npx truffle develop
truffle(develop)> test
...
```
. However, this does not work for me. 

I use the command `truffle test` in the folder that I cloned from the tutorial. I got this command  from `https://trufflesuite.com/tutorial/index.html#running-the-tests`.

In addition, I tried add the `.js` test files and run `truffle test`. It still show me the raining characters.

Did I still miss something? @JoepdeJong

**Comment by JoepdeJong** [comment]
> Do we have to use a combination of `.sol` and `.js` test files for testing the smart contract?

There is no need to have `.js` files for testing the smart contract. Testing with only `.sol` is supposed to work too.

What does `npm list colors` output when you run it in the folder of your project? @plainfatboy

**Comment by plainfatboy** [comment]
@JoepdeJong 

The output of `npm list colors` is as following

```
pet-shop@1.0.0 C:\Users\Kiki\first-smart-contract
`-- colors@1.4.0
```

@JoepdeJong

**Comment by JoepdeJong** [comment]
> @JoepdeJong
> 
> The output of `npm list colors` is as following
> 
> ```
> pet-shop@1.0.0 C:\Users\Kiki\first-smart-contract
> `-- colors@1.4.0
> ```

Try installing `truffle` as a dependency of your project using `npm install truffle`. In this way, the `truffle` of your project overrides your global installed `truffle`, and uses the version `1.4.0` of `colors`. @plainfatboy

**Comment by plainfatboy** [comment]
@JoepdeJong I tried `npm install truffle` and the output of `npm list colors` is as following

```
pet-shop@1.0.0 C:\Users\Kiki\first-smart-contract
+-- colors@1.4.0
`-- truffle@5.4.27
  `-- @truffle/debugger@9.2.9
    `-- @truffle/codec@0.11.21
      `-- @truffle/compile-common@0.7.23
        `-- colors@1.4.0 deduped
```

However, it still shows the raining characters.

**Comment by JoepdeJong** [comment]
Did you use `npx truffle test`? or `truffle test`? @plainfatboy

**Comment by plainfatboy** [comment]
I am sorry. I forgot to use `npx truffle test`. Now, it works by using `npx truffle test`. Thank you very much. @JoepdeJong .

However, I still wonder why the global installed `truffle` does not work.

**Comment by JoepdeJong** [comment]
> However, I still wonder why the global installed `truffle` does not work.

This is because someone (on purpose) broke a dependency. See https://github.com/Marak/colors.js/issues/285

When this issue is fixed you can use the latest version of `colors` again. I am following the progress so I will let you know when this is the case!

**Comment by gnidan** [comment]
Thanks all! Really appreciate the issue reports here. This has been fixed and released in [Truffle v5.4.28](https://github.com/trufflesuite/truffle/releases/tag/v5.4.28). Please upgrade!
